### PR TITLE
Support for compilation with clang

### PR DESCRIPTION
**Pull request by raoulstrackx**
Clang does not know every GCC warning (e.g., -Wmaybe-uninitialized), the necessary checks were added.

### CODE REVIEW

**Comment by jethrogb** [comment]
Shouldn't you add the same in UnwindLevel1-gcc-ext.c, UnwindRustSgxSnprintf.c, UnwindLevel1.c?

**Comment by raoulstrackx** [comment]
Well it works without :) But yes I've been a bit too fast with the commit. I'll do a complete check.

File: libunwind/src/UnwindRustSgxSnprintf.c [hunk 1]
```diff
@@ -5,11 +5,12 @@
  * on all source code distributions
  */
 
+#pragma GCC diagnostic ignored "-Wunknown-pragmas"
 #pragma GCC diagnostic ignored "-Wconversion"
-#pragma GCC diagnostic ignored "-Wdiscarded-qualifiers"
```
Comment (jethrogb) [inline libunwind/src/UnwindRustSgxSnprintf.c:9]: Don't delete this

File: libunwind/src/UnwindLevel1.c [hunk 2]
```diff
@@ -19,7 +19,7 @@
 // to export these functions from libunwind.so as well.
 #define _LIBUNWIND_UNWIND_LEVEL1_EXTERNAL_LINKAGE 1
 
-#pragma GCC diagnostic ignored "-Wunused-but-set-variable"
```
Comment (jethrogb) [inline libunwind/src/UnwindLevel1.c:22]: Don't delete this

File: libunwind/src/UnwindLevel1.c [hunk 3]
```diff
@@ -19,7 +19,7 @@
...
+#pragma GCC diagnostic ignored "-Wunknown-pragmas"
```
Comment (jethrogb) [inline libunwind/src/UnwindLevel1.c]: Does this work in GCC as well?

File: libunwind/src/UnwindLevel1.c [hunk 4]
```diff
@@ -69,13 +69,15 @@ unwind_phase1(unw_context_t *uc, unw_cursor_t *cursor, _Unwind_Exception *except
 
     // When tracing, print state information.
     if (_LIBUNWIND_TRACING_UNWINDING) {
+#ifndef NDEBUG
```
Comment (jethrogb) [inline libunwind/src/UnwindLevel1.c]: What's this for?

File: libunwind/src/libunwind.cpp [hunk 5]
```diff
@@ -10,7 +10,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
```
Comment (jethrogb) [inline libunwind/src/libunwind.cpp:13]: Don't comment this

File: libunwind/src/CMakeLists.txt [hunk 6]
```diff
@@ -121,7 +125,7 @@ string(REPLACE ";" " " LIBUNWIND_LINK_FLAGS "${LIBUNWIND_LINK_FLAGS}")
 set_property(SOURCE ${LIBUNWIND_CXX_SOURCES}
              APPEND_STRING PROPERTY COMPILE_FLAGS " ${CMAKE_CXX_FLAGS} ${LIBUNWIND_CXX_FLAGS}")
```
Comment (jethrogb) [inline libunwind/src/CMakeLists.txt]: Delete this one as well then?

**Pull request merged**